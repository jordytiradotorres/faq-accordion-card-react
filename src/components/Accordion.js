import AccordionItem from "./AccordionItem";
import "./styles/Accordion.css";

const Accordion = () => {
  const details = [
    {
      id: 1,
      title: "How many team members can I invite?",
      description:
        "You can invite up to 2 additional users on the Free plan. There is no limit on team members for the Premium plan.",
    },
    {
      id: 2,
      title: "What is the maximum file upload size?",
      description:
        "No more than 2GB. All files in your account must fit your allotted storage space.",
    },
    {
      id: 3,
      title: "How do I reset my password?",
      description:
        "Click “Forgot password” from the login page or “Change password” from your profile page. A reset link will be emailed to you.",
    },
    {
      id: 4,
      title: "Can I cancel my subscription?",
      description:
        "Yes! Send us a message and we’ll process your request no questions asked.",
    },
  ];

  return (
    <section className="accordion">
      <div className="accordion__img">
        <picture>
          <source
            srcSet={
              process.env.PUBLIC_URL +
              "/assets/images/illustration-woman-online-desktop.svg"
            }
            media="(min-width: 768px)"
          />
          <img
            src={
              process.env.PUBLIC_URL +
              "/assets/images/illustration-woman-online-mobile.svg"
            }
            alt="illustration woman"
          />
        </picture>
        <img
          className="img-arroba"
          src={
            process.env.PUBLIC_URL +
            "/assets/images/illustration-box-desktop.svg"
          }
          alt="pattern"
        />
      </div>

      <div className="accordion__content">
        <h2>FAQ</h2>

        {details.map((detail) => (
          <AccordionItem key={detail.id} {...detail} />
        ))}
      </div>
    </section>
  );
};

export default Accordion;
