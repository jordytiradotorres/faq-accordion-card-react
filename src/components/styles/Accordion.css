.accordion {
  background-color: #ffffff;
  border-radius: 1rem;
  padding: 3rem 2rem;
  margin-top: 10rem;
  width: 305px;
}
@media (min-width: 1024px) {
  .accordion {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    margin-top: 0;
    padding-right: 8rem;
    padding-left: 0;
    padding-bottom: 0;
    padding-top: 0;
  }
}
.accordion__img {
  margin-top: -12rem;
  position: relative;
}
@media (min-width: 1024px) {
  .accordion__img {
    margin-top: 0;
    width: 400px;
  }
}
.accordion__img img {
  background-image: url(./../../images/bg-pattern-mobile.svg);
  background-position: top 0% left 44%;
  background-size: cover;
}
@media (min-width: 1024px) {
  .accordion__img img {
    background-image: url(./../../images/bg-pattern-desktop.svg);
    padding: 9rem 0;
    background-position: right -1% top 68%;
    background-size: initial;
    margin-left: -6rem;
  }
}
.accordion__img .img-arroba {
  display: none;
}
@media (min-width: 1024px) {
  .accordion__img .img-arroba {
    display: block;
    position: absolute;
    top: 8rem;
    left: -2rem;
    z-index: 100;
  }
}
.accordion__content h2 {
  text-align: center;
  font-size: 2rem;
  margin: 2.5rem 0;
  color: #1d1e35;
}
@media (min-width: 1024px) {
  .accordion__content {
    padding-bottom: 5rem;
  }
}
@media (min-width: 1024px) {
  .accordion__item {
    width: 330px;
  }
}
.accordion__item .content-show {
  display: flex;
  justify-content: space-between;
  padding: 0.7rem 0;
  cursor: pointer;
}
.accordion__item .content-show h3 {
  font-weight: 300;
  font-size: 1rem;
  transition: color 0.2s ease-in-out;
}
.accordion__item .content-show h3:hover {
  color: #f47c57;
}
@media (min-width: 1024px) {
  .accordion__item .content-show h3 {
    font-size: 1.1rem;
  }
}
.accordion__item .content-show button {
  border: none;
  background: transparent;
  cursor: pointer;
}
.accordion__item p {
  padding: 0.5rem 0;
  color: #787887;
  line-height: 1.4;
  display: none;
  transition: all 3s ease-in-out;
}
.accordion__item .show {
  display: block;
}
