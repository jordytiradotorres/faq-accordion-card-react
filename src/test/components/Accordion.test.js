import {shallow} from "enzyme";
import Accordion from "../../components/Accordion";

describe("Pruebas en el archivo Accordion.js", ()=> {
    const wrapper = shallow(<Accordion />)

    test("Debe de renderizar correctamente", () => {
        expect(wrapper).toMatchSnapshot()
    })

    test("Debe de mostrar 4 elementos AccordionItem", () => {
        expect(wrapper.find("AccordionItem").length).toBe(4)
    })

    test("Debe de mostrar la imagen correcta", () => {
        const url = "/assets/images/illustration-box-desktop.svg"
        const imgSrc = wrapper.find(".img-arroba").prop("src")

        expect(url).toBe(imgSrc)
    })
})