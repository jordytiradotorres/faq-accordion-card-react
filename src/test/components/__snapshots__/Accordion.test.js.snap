// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo Accordion.js Debe de renderizar correctamente 1`] = `
<section
  className="accordion"
>
  <div
    className="accordion__img"
  >
    <picture>
      <source
        media="(min-width: 768px)"
        srcSet="/assets/images/illustration-woman-online-desktop.svg"
      />
      <img
        alt="illustration woman"
        src="/assets/images/illustration-woman-online-mobile.svg"
      />
    </picture>
    <img
      alt="pattern"
      className="img-arroba"
      src="/assets/images/illustration-box-desktop.svg"
    />
  </div>
  <div
    className="accordion__content"
  >
    <h2>
      FAQ
    </h2>
    <AccordionItem
      description="You can invite up to 2 additional users on the Free plan. There is no limit on team members for the Premium plan."
      id={1}
      key="1"
      title="How many team members can I invite?"
    />
    <AccordionItem
      description="No more than 2GB. All files in your account must fit your allotted storage space."
      id={2}
      key="2"
      title="What is the maximum file upload size?"
    />
    <AccordionItem
      description="Click “Forgot password” from the login page or “Change password” from your profile page. A reset link will be emailed to you."
      id={3}
      key="3"
      title="How do I reset my password?"
    />
    <AccordionItem
      description="Yes! Send us a message and we’ll process your request no questions asked."
      id={4}
      key="4"
      title="Can I cancel my subscription?"
    />
  </div>
</section>
`;
